File: Algorithms/137/single-number-ii.c
int singleNumber(int* nums, int numsSize){
    int a = 0, b = 0;
    for (int i = 0; i < numsSize; ++i) {
        a = (a^nums[i]) & ~b;
        b = (~a & nums[i]) ^ b;
    }
    return a;
}

File: Algorithms/137/single-number-ii.go
package main

import (
	"fmt"
	"sort"
)

func main() {
	ary := []int{-2, -2, 1, 1, -3, 1, -3, -3, -4, -2}

	fmt.Println(singleNumber(ary))
}

func singleNumber(nums []int) int {
	sort.Ints(nums)
	for i, v := range nums {
		lst := v - 1
		if i-1 > 0 {
			lst = nums[i-1]
		}

		nxt := v + 1
		if i+1 < len(nums) {
			nxt = nums[i+1]
		}

		if lst != v && nxt != v {
			return v
		}
	}
	return 0
}
